src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1563,6 +1581,10 @@ void TorrentImpl::endReceivedMetadataHandling(const Path &savePath, const PathLi
     p.ti = metadata;
     p.file_priorities = filePriorities;
 
+    // Check file name blacklist when metadata was not available in Session::addTorrent_impl
+    if (!hasFilePriorities)
+        applyExcludedFileNamesFilters();
```

Comment:
It is still done incorrectly.

Reply:
All you need is replace
```c++
const auto priority = LT::fromNative(filePriorities[LT::toUnderlyingType(nativeIndex)]);
```
with something like the following in loop above:
```c++
        lt::download_priority_t &nativePriority = filePriorities[LT::toUnderlyingType(nativeIndex)];
        if (nativePriority != lt::dont_download) && m_session->isFilenameExcluded(this->filePath(i).filename()))
            nativePriority = lt::dont_download;
        const auto priority = LT::fromNative(nativePriority);
```

Reply:
Like this?

```diff
diff --git a/src/base/bittorrent/torrentimpl.cpp b/src/base/bittorrent/torrentimpl.cpp
index f9a3fb24f..37b601167 100644
--- a/src/base/bittorrent/torrentimpl.cpp
+++ b/src/base/bittorrent/torrentimpl.cpp
@@ -1553,7 +1553,6 @@ void TorrentImpl::endReceivedMetadataHandling(const Path &savePath, const PathLi
     Q_ASSERT(m_filePaths.isEmpty());
 
     lt::add_torrent_params &p = m_ltAddTorrentParams;
-    const bool hasFilePriorities = m_ltAddTorrentParams.file_priorities.size() != 0;
 
     const std::shared_ptr<lt::torrent_info> metadata = std::const_pointer_cast<lt::torrent_info>(nativeTorrentInfo());
     m_torrentInfo = TorrentInfo(*metadata);
@@ -1574,17 +1573,22 @@ void TorrentImpl::endReceivedMetadataHandling(const Path &savePath, const PathLi
 
         m_filePaths.append(filePath.removedExtension(QB_EXT));
 
-        const auto priority = LT::fromNative(filePriorities[LT::toUnderlyingType(nativeIndex)]);
+        lt::download_priority_t nativePriority = filePriorities[LT::toUnderlyingType(nativeIndex)];
+
+        // Check file name blacklist when metadata was not available in Session::addTorrent_impl
+        if (nativePriority != lt::dont_download && m_session->isFilenameExcluded(filePath.filename()))
+        {
+            nativePriority = lt::dont_download;
+        }
+
+        const auto priority = LT::fromNative(nativePriority);
+
         m_filePriorities.append(priority);
     }
     p.save_path = savePath.toString().toStdString();
     p.ti = metadata;
     p.file_priorities = filePriorities;
 
-    // Check file name blacklist when metadata was not available in Session::addTorrent_impl
-    if (!hasFilePriorities)
-        applyExcludedFileNamesFilters();
-
     reload();
 
     // If first/last piece priority was specified when adding this torrent,

```

Reply:
> Like this?

You missed a couple of essential details from my proposal, and also broke the coding style a little.

Reply:
Fixed it myself.